StacksOfPlates.py

```diff
@@ -0,0 +1,48 @@
+class Stack:
+    def __init__(self, cap):
+        self.stack = []
+        self.cap = cap
+
+    def isFull(self):
+        return len(self.stack) == self.cap
+
+    def push(self, item):
+        if not self.isFull():
+            self.stack.append(item)
+
+    def pop(self):
+        if self.stack:
+            return self.stack.pop()
+
+class SetOfStacks:
+    def __init__(self, stackCap):
+        self.stacks = []
+        self.stackCap = stackCap
+
+    def push(self, item):
+        if not self.stacks or self.stacks[-1].isFull():
+            newStack = Stack(self.stackCap)
+            newStack.push(item)
+            self.stacks.append(newStack)
+        else:
+            self.stacks[-1].push(item)
```

Comment:
This could be done more cleanly by always running line 28, and only generating the new stack if line 23 evaluates to true.
```suggestion
        if not self.stacks or self.stacks[-1].isFull():
            self.stacks.append(Stack(self.stackCap))
        self.stacks[-1].push(item)
```